projects/103_homework/src/main.c

```diff
@@ -0,0 +1,60 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+// interrupt call back function
+void interrupt_callback(const GpioAddress *address, void *context) {
```

Comment:
If a function is used only in the file it is declared in, ie by other methods, it is important to declare it static and prefix it with `prv_` so that we can know that it is a 'private function', only visible in the scope of the program. The concept of static in C allows us to better structure our code, and retain the scope of any function to only that which it needs. See [here](https://www.tutorialspoint.com/static-functions-in-c)